components/apimgt/org.wso2.carbon.apimgt.throttling.siddhi.extension/src/test/java/org/wso2/carbon/apimgt/throttling/siddhi/extension/CountAttributeAggregatorWithResetTestCase.java

```diff
@@ -0,0 +1,203 @@
+                "insert all events into outputStream;";
+
+        ExecutionPlanRuntime execPlanRunTime = siddhiManager.createExecutionPlanRuntime(execPlan);
+
+        execPlanRunTime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                if (inEvents != null) {
+                    atomicEventCount.addAndGet(inEvents.length);
+                    if (atomicEventCount.get() == 1) {
+                        junit.framework.Assert.assertEquals(3L, inEvents[0].getData(0));
+                    }
+                }
+            }
+        });
+
+        execPlanRunTime.start();
+        InputHandler inputHandler = execPlanRunTime.getInputHandler("cseEventStream");
+        inputHandler.send(new Object[]{"WSO2", 0F});
+        Thread.sleep(1000);
+        inputHandler.send(new Object[]{"WSO2", 0F});
+        Thread.sleep(1000);
+        inputHandler.send(new Object[]{"APIM", 3F});
+        Thread.sleep(1000);
+        inputHandler.send(new Object[]{"WSO2", 3F});
+        Thread.sleep(1000);
+        inputHandler.send(new Object[]{"APIM", 3F});
+        Thread.sleep(2000);
+        inputHandler.send(new Object[]{"APIM", 3F});
+        execPlanRunTime.shutdown();
+        junit.framework.Assert.assertEquals(1, atomicEventCount.intValue());
+    }
+
+    @Test
+    public void CountAggregatorTest3() throws InterruptedException {
+        log.info("CountAggregator Test #3 : Setting Reset value to be true");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String execPlan = "" +
+                "@Plan:name('CountAggregatorTests') " +
+                "" +
+                "define stream cseEventStream (symbol string, price float);" +
+                "" +
+                "@info(name = 'query1') " +
+                "from cseEventStream#window.timeBatch(2 sec) " +
+                "select symbol, ifThenElse(price == -1F, throttler:count(price, true),throttler:count(price, false)) as count " +
+                "group by symbol " +
+                "insert all events into outputStream;";
+
+
+        ExecutionPlanRuntime execPlanRunTime = siddhiManager.createExecutionPlanRuntime(execPlan);
+
+        execPlanRunTime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                atomicEventCount.set(0);
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                if (inEvents != null) {
+                    atomicEventCount.addAndGet(inEvents.length);
+                    if (atomicEventCount.get() == 1) {
+                        junit.framework.Assert.assertEquals("APIM", inEvents[0].getData(0));
+                        junit.framework.Assert.assertEquals(1L, inEvents[0].getData(1));
+
```

Comment:
Code not formatted. Empty lines are there

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12443/commits/a583c6ae8e5be9a7a7879183801ad442b86123e4#diff-a92a928154247caabe1158c2ac06adfeab5798b6f9e50538cb203bffabb50a70